go/ql/test/library-tests/semmle/go/frameworks/Fasthttp/fasthttp.go

```diff
@@ -20,33 +27,42 @@ func fasthttpClient() {
 
 	res := &fasthttp.Response{}
 	req := &fasthttp.Request{}
-	req.SetHost(userInput)                // $ hasTaintFlow="req" ReqPred=userInput
-	req.SetHostBytes(userInputByte)       // $ hasTaintFlow="req" ReqPred=userInputByte
-	req.SetRequestURI(userInput)          // $ hasTaintFlow="req" ReqPred=userInput
-	req.SetRequestURIBytes(userInputByte) // $ hasTaintFlow="req" ReqPred=userInputByte
+	req.SetHost(source())
+	sink(req) // $ hasTaintFlow="req"
+	req.SetHostBytes(userInputByte)
+	sink(req) // $ hasTaintFlow="req"
```

Comment:
The tests look better. The only problem is that line 33 isn't really checking anything about line 32 because we've established from line 31 that `req` is tainted. You need to use different sources for each of these "source to sink" tests.

Reply:
When we run in CI we run the tests with an extra flag `--consistency-queries go/ql/test/consistency` which also runs the consistency checks. At the moment all they do is check that there weren't any build errors when we ran the extractor.